### PR TITLE
Created "and" rule

### LINKED ISSUE #10: Create AND rule
Create AND rule. Reference: https://github.com/paritytech/pr-custom-review#and-rule-syntax-

### PR DESCRIPTION
**Pull request by Bullrich**
# Changes
- Created the `AND` rule with a configuration.
  - This resolves #10 
  - Created the type for it
  - Created the JOI validation
    - It must have at least two individual conditions for it to be valid so it is not used as a `basic` rule.
- Fixed bug where check summary duplicated content
  - This happened because it has a buffer and I had to empty it.
- Created `AND` rule evaluation for pull requests.
  - It combines its report into one.
- Fixed bug where author appeared as one of the missing reviews
- Created tests for `AND` rule.
- Renamed files and directories
  - `file` directory now is `rule` directory (as it store the types and validations for the clases)

### CODE REVIEW

**Review by mutantcornholio — APPROVED** [review]
testing testing

**Comment by Bullrich** [comment]
@paritytech/opstooling this is ready for review. I had **not** removed the branch code so you can see how it works on its own branch.

Will be removed at the end (anyways, it is a requirement to make `validate-action` step success.

File: src/test/runner/validation/and.test.ts [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+            name: "And rule",
+            type: RuleTypes.And,
+            condition: { include: ["review-bot.yml"] },
+            reviewers: [{ teams: ["abc"], min_approvals: 1 }],
+          },
+        ],
+      };
+      api.listApprovedReviewsAuthors.mockResolvedValue([users[0]]);
+      const evaluation = await runner.validatePullRequest(config);
+      expect(evaluation).toHaveLength(0);
```
Comment (mutantcornholio) [inline src/test/runner/validation/and.test.ts:39]: > should return true if the reviewers reviewed

> `      expect(evaluation).toHaveLength(0);`

Feels like the `return true` part is outdated

File: .github/workflows/review-bot.yml [hunk 2]
```diff
@@ -26,7 +26,7 @@ jobs:
           private_key: ${{ secrets.TEAM_APP_KEY }}
       # !This must always point to main. 
       # Change it for the PRs but remember to change it back
-      - uses: paritytech/review-bot@main
+      - uses: paritytech/review-bot@and-rule
```
Comment (mutantcornholio) [inline .github/workflows/review-bot.yml:29]: leftover?

**Review by mutantcornholio — DISMISSED** [review]

File: .github/workflows/review-bot.yml (hunk 2 above)
Comment (Bullrich) [reply .github/workflows/review-bot.yml:29]: Intended for now so you can see how it works. Will fix this when the approvals have been met.

File: src/test/runner/validation/and.test.ts (hunk 1 above)
Comment (Bullrich) [reply src/test/runner/validation/and.test.ts]: image

**Review by rzadp — APPROVED** [review]

**Pull request merged**